* update the datatables library to 1.9.4 as 1.8.2 does not support "aoColumndefs"

apremalal opened the pull request:
- updated the reference UI client search tab with server side pagination support.
- pagination is ready to add to all the UI components,implemented in a customizable way where,we can define table column templates and the corresponding json  elements.

*Another performance improvement is that we can request only the data needed to render the table instead of all the column details.

Commit:
* update the datatables library to 1.9.4 as 1.8.2 does not support "aoColumndefs"
* updated the reference UI client search tab with server side pagination support.

keithwoodlock commented:
Anuruddha,

Overall, I really liked this below are my comments. Not bring in work into repository just yet. 

I like that you removed the 'dropdown' list on clients search page and replaced with paginated approach through latest datatables.
I like that you were able to retain the 'office data scoping' within the datatables widget (how did you do that? :) )

Some bits and pieces dont seem 100% right at present - if they were I would just bring in this work straightaway cause its def an improvement over the older clients search screen.

example request from datatables widget: /clients/paginated?offset=1&limit=10&_=1367317445093
- offset starts at 1 instead of 0 cutting out one search result 
- You pass in the limit used from datatables yet I still see text in search results of 'filtered from 200 total entries' where is it getting the 200 from? Is this the hard limit on backend? if so shouldnt this just be changed to requested limit if smaller?
- entering text is search box doesnt work correctly - i see the underlying api call do the right thing: 

``` https://localhost:8443/mifosng-provider/api/v1/clients/paginated?offset=1&limit=10&sqlSearch=display_name+like+%27%25one%25%27&underHierarchy=.&_=1367318276848```

yet it does not return back clients which should match against entered text.

Suggested improvement
  - Is it possible to supply other page sizes other than 10, 25, 50 & 100 in datatables widget
  - Would it be possible to set the default a bit higher at 100 maybe



```

apremalal commented:
Keith,

I really appreciate your feedback.

I've made a mistake in adding a 1 to the offset parameter (in the FE) and it caused the search issue.After removing that +1 everything works like a charm.

But one thing remains to fix that is 'filtered from 200 total entries' ,i've tried so hard on that.It basically tells how many rows present in the DB ,which in our case unnecessary(as it would add more cost). I've set that variable to 200 in FE code.

In datatable plugin we have an option to set what should be shown in the information bar "oLanguage":,
It's not working for some reason.I thought of posting the issue to datatables forum.Will fix that as soon as I can.

Other than that I've made all the changes and improvements you've mentioned.

In page sizes what would be ideal,Currently I've changed it to [200, 150, 100, 50]?.

Hope to make the pull request as soon as I fixed the information bar issue.

PS: regarding external html elements :), yes it was a bit hard to find-out a way in which to inject html elements to the datatables layout. I've achieved it by dynamically binding HTML element on the rendering event of the table,then bond the corresponding event to call "fnDraw()".
